crates/ty_python_semantic/src/types/instance.rs

```diff
@@ -132,6 +133,16 @@ impl<'db> Type<'db> {
         relation_visitor: &HasRelationToVisitor<'db>,
         disjointness_visitor: &IsDisjointVisitor<'db>,
     ) -> ConstraintSet<'db> {
+        // Iff we're checking a NamedTuple against NamedTupleLike, skip the `_replace` method
+        // signature. NamedTuples synthesize `_replace` methods with specific keyword-only
+        // parameters (to detect invalid arguments), which are not strictly subtypes of the
+        // protocol's `(**kwargs)` signature, but are intended to be considered as satisfying it.
+        let is_namedtuple_protocol_check = matches!(&protocol.inner, Protocol::FromClass(class) if class.is_known(db, KnownClass::NamedTupleLike))
+            && self.as_nominal_instance().is_some_and(|instance| {
+                let (class_literal, specialization) = instance.class(db).class_literal(db);
+                CodeGeneratorKind::NamedTuple.matches(db, class_literal, specialization)
+            });
```

Comment:
I guess for the mypy equivalent we'd just totally short-circuit here and return true, since we've already done the work to check that we're comparing a named-tuple with `NamedTupleLike`. That seems... probably fine in practice? I'd want to see if @AlexWaygood knows of any reason we shouldn't just do that.